ivy-prescient.el

```diff
@@ -111,19 +111,33 @@ This is for use in `ivy-re-builders-alist'."
 (defvar ivy-prescient--old-re-builder nil
   "Previous default value in `ivy-re-builders-alist'.")
 
+(defun ivy-prescient--elements-ensure (element)
+  "Ensure that the type of `ELEMENT' is acceptable to prescient.
+In addition to string, ivy accepts many types of data, so some
+processing is needed to ensure that `prescient' can handle them."
+  (cond ((symbolp element) (symbol-name element))
+        ((consp element) (symbol-name (car element)))
+        ((listp element) (car element))))
```

Comment:
Can we update this function so that it is safe to call even if `ELEMENT` is already a string? That would simplify the calling code, and prevent a future bug where somebody does not add the `stringp` guard.